chain/rust/src/builders/tx_builder.rs

```diff
@@ -227,10 +228,54 @@ fn min_fee(tx_builder: &TransactionBuilder) -> Result<Coin, TxBuilderError> {
 fn min_fee_with_exunits(tx_builder: &TransactionBuilder) -> Result<Coin, TxBuilderError> {
     let full_tx = fake_full_tx(tx_builder, tx_builder.build_body()?)?;
     // we can't know the of scripts yet as they can't be calculated until we build the tx
+    
+    fn ref_script_orig_size_builder(utxo: &TransactionUnspentOutput) -> Option<(ScriptHash, u64)> {
+        utxo.output.script_ref().map(|script_ref| (script_ref.hash(), script_ref.raw_plutus_bytes().expect("TODO: handle this").len() as u64))
+    }
+    // let ref_script_orig_sizes: std::collections::BTreeMap<ScriptHash, u64> = if let Some(ref_inputs) = &tx_builder.reference_inputs {
+    //     ref_inputs.iter().chain(tx_builder.inputs.iter()).filter_map(ref_script_orig_size_builder).collect()
+    // } else {
+    //     tx_builder.inputs.iter().filter_map(ref_script_orig_size_builder).collect()
+    // };
+
+    // let mut total_ref_script_size = 0;
+    // for output in tx_builder.outputs.iter() {
+    //     if let Some(Credential::Script{ hash, .. }) = output.address().payment_cred() {
+    //         if let Some(orig_size) = ref_script_orig_sizes.get(hash) {
+    //             total_ref_script_size += *orig_size;
+    //             println!("USING REF SCRIPT {} | SIZE {}", hash.to_hex(), orig_size);
+    //         }
+    //     }
+    // }
+
+    let ref_script_orig_sizes: HashMap<ScriptHash, u64> = if let Some(ref_inputs) = &tx_builder.reference_inputs {
```

Comment:
From reading CIP33 it says:
```
The key idea is to use reference inputs and modified outputs which carry actual scripts ("reference scripts"), and allow such reference scripts to satisfy the script witnessing requirement for a transaction. This means that the transaction which uses the script will not need to provide it at all, so long as it referenced an output which contained the script.
```

so I'm going on the assumption that if the hash matches something created as a reference script in a reference input's output that it must be a reference script and will incur this extra fee

other option: be explicit during tx builder/input builder and get the info from there